cmd/evm/internal/t8ntool/execution.go

```diff
@@ -206,7 +206,12 @@ func (pre *Prestate) Apply(vmConfig vm.Config, chainConfig *params.ChainConfig,
 			rejectedTxs = append(rejectedTxs, &rejectedTx{i, errMsg})
 			continue
 		}
-		msg, err := core.TransactionToMessage(tx, signer, pre.Env.BaseFee, vmContext.ExchangeRates)
+		// NOTE: we can't provide exchange rates
+		// for fee-currencies here, since those are dynamically changing
+		// based on the oracle's exchange rates.
+		// When a Celo transaction with specified fee-currency is validated with this tool,
+		// this will thus result in a ErrNonWhitelistedFeeCurrency error for now.
+		msg, err := core.TransactionToMessage(tx, signer, pre.Env.BaseFee, vmContext.FeeCurrencies.ExchangeRates)
```

Comment:
Is anyone using this tool? Do we want to support this for fee-currency transactions?
How does the user set the exchange rate values to be used for the state simulation?